client/scala/src/main/scala/ai/verta/client/ClientConnection.scala

```diff
@@ -1,7 +1,16 @@
 package ai.verta.client
 
+import java.net.URI
+
 case class ClientConnection(host: String, ignoreConnErr: Boolean = false, maxRetries: Int = 5, auth: ClientAuth = null)
 
 object ClientConnection {
-  def fromEnvironment(): ClientConnection = new ClientConnection(sys.env("VERTA_HOST"), auth = ClientAuth.fromEnvironment())
+  private val DefaultScheme = "https"
+
+  private def hasScheme(url: String) = (new URI(url)).getScheme() != null
+
+  private def urlWithScheme(url: String) =
+    if (hasScheme(url)) url else (new URI(DefaultScheme, url, null, null, null)).toString()
```

Comment:
Could we just replace the scheme? This is making hte assumption that 3 parts of the URI are `null`, which seems like a stronger assumption than we have to make here.

Reply:
I think the most general solution is to pass the URL to the URI constructor, then use URI's method to extract the various components. Something like this:
```scala
val uri = new URI(url)

(new URI(Option(uri.getScheme()).getOrElse(DefaultScheme), uri.getAuthority(), uri.getPath(), uri.getQuery(), uri.getFragment())
```
Or one of the other constructors, which is probably a bit less general.

Reply:
See new commit 🤔

Reply:
Actual neither is my original attempt =) It probably picks up localhost as scheme =)

Reply:
Localhost as scheme is crazy!